en/middleware/micrortps.md

```diff
@@ -0,0 +1,344 @@
+# RTPS/ROS2 interface: the PX4-FastRTPS bridge
+
+This bridge adds communication capabilities between a *PX4 Autopilot* and a *Fast RTPS* application, through serial ports or
+UDP sockets, using [CDR serialization](https://en.wikipedia.org/wiki/Common_Data_Representation). The goal is to provide a DDS (Data Distribution Service) interface to PX4. This interface will also allow sharing information with the forthcoming release of ROS2 (Robot Operating System).
+
+RTPS is the underlying protocol of DDS, an standard from the OMG (Object Management Group) providing a real-time pub/sub middleware widely used in aerospace, defense and IoT applications, while in Robotics has been adopted as the middleware for ROS2.
+
+Fast RTPS implements the latest version of the RTPS protocol and a minimum DDS API, resulting in a very lightweight implementation of the standard and full access to the RTPS fine settings.
+
+![micro RTPS general scheme](../../assets/middleware/micrortps/micrortps_general_scheme.png)
+
+## Code generation
+
+Support for the functionality is mainly implemented within three new (automatically generated) code blocks:
+
+- *CDR serialization functions* are generated for specified uORB topics (those that are to be sent/received). For example, the following functions are generated for the *sensor_combined.msg*:
+
+  ```sh
+  void serialize_sensor_combined(const struct sensor_combined_s *input, char *output, uint32_t *length, struct microCDR *microCDRWriter);
+  void deserialize_sensor_combined(struct sensor_combined_s *output, char *input, struct microCDR *microCDRReader);
+  ```
+  
+  > **Note** *CDR serialization* provides a common format for exchanging serial data between different platforms.  
+
+- A *client application* that sends and receives the CDR serialized data from topics through a selected UART or UDP port.
+
+  ![micro RTPS client](../../assets/middleware/micrortps/micrortps_client.png)
+
+- An *agent application* that shares the CDR serialized data via (using **Fast RTPS**) to DDS service clients (e.g. ROS2). An important step in this part is the generation of a IDL file for each topic which is the translation of the corresponding MSG file. For the case of *sensor_combined* topic the **sensor_combined_.idl** file is generated from **sensor_combined.msg** file.
```

Comment:
I have changed "important" for "first" for pointing that is an clarification over the process.